arviz/data/io_pymc3.py

```diff
@@ -153,18 +155,30 @@ def _extract_log_likelihood(self):
         """Compute log likelihood of each observation."""
         # If we have predictions, then we have a thinned trace which does not
         # support extracting a log likelihood.
-        cached = [(var, var.logp_elemwise) for var in self.model.observed_RVs]
-        log_likelihood_dict = {}
+        if self.log_likelihood is True:
+            cached = [(var, var.logp_elemwise) for var in self.model.observed_RVs]
+        else:
+            cached = [
+                (var, var.logp_elemwise)
+                for var in self.model.observed_RVs
+                if var.name in self.log_likelihood
+            ]
+        try:
+            log_likelihood_dict = self.pymc3.sampling._DefaultTrace(  # pylint: disable=protected-access
+                len(self.trace.chains)
+            )
+        except AttributeError:
+            raise AttributeError(
+                "Either upgrade PyMC3 to latest version or downgrade ArviZ for log_likelihood."
```

Comment:
"Installed version of ArviZ requires PyMC3>=3.8. Please upgrade with `pip install pymc3>=3.8` "
    "or `conda install -c conda-forge pymc3>=3.8`."